### ISSUE TITLE
[BUG] Problem copying a plot from one jupyter cell to another

### ISSUE THREAD

**Issue by axil** [bokeh/bokeh#12194]
### Software versions

Python version      :  3.7.7 (tags/v3.7.7:d7c567b08f, Mar 10 2020, 10:41:24) [MSC v.1900 64 bit
 (AMD64)]
IPython version     :  7.30.1
Tornado version     :  6.0.3
Bokeh version       :  2.4.3
BokehJS static path :  C:\Python37-64\lib\site-packages\bokeh\server\static
node.js version     :  (not installed)
npm version         :  (not installed)
Operating system    :  Windows-10-10.0.19041-SP0

### Expected behavior

When I copy-paste a cell with a plot with Ctrl-C, Ctrl-V, the plot is copied to a new cell

### Observed behavior

The plot is duplicated in the cell being copied and does not appear in the target cell

### Example code

```Python
from bokeh.plotting import figure, output_file, show
p = figure()
x = [1,2,3]
y = [3,5,4]
p.scatter(x, y)
show(p)
```

### Screenshots

![image](https://user-images.githubusercontent.com/170910/175605398-eff9e0e0-82da-4337-b7a8-41e1a442d3ee.png)

**Comment by bryevdv** [comment]
Bokeh plots are actually JavaScript programs. I would not expect this to work, and to be honest I am not sure it can ever be made to work. 

cc @bokeh/dev

**Comment by axil** [comment]
Matplotlib plots are copied correctly, so it is sort of expected that bokeh also does something 'right' when copy-pasting.

I would not be surprised if the new cell would not display any output and that would be it. But duplicating the output of the old cell looks like a bug.

**Comment by bryevdv** [comment]
@axil Matplotlib plots are static images, Bokeh plots are interactive JavaScript programs. Can you clarify your exact sequence of operations?  I can think of at least two different interpretations of "copying" and I want to make sure we are talking about the exact same thing.

**Comment by axil** [comment]
(I tried all kind of copy-pastes, but this one is the simplest) I press "C" keyboard button on a cell, then I press "V".

**Comment by bryevdv** [comment]
@axil Can you confirm that copying the *text* in the cell (not the "cell itself") and pasting that text into a new cell then executing the new cell works as expected?

**Comment by axil** [comment]
Yes. I confirm that. That's what I have to do every time I need to copy-paste a cell with a plot.

Also, I think I've found the problem. It is in the 'id' attribute of the 'bk-root' div. If I delete it, copy-pasting works as expected.

**Comment by bryevdv** [comment]
Right, I expect the issue is that Jupyter copies the output cell in some way or other. But embedding Bokeh plots requires specifying a div id to embed into. So if jupyter just copies the output as-is, then it embeds into the previous div (which is in the output cell).

The only only hope of making this work is if Jupyter provides some sort of hook that can customize the copy behaviour. I expect if that is possible at all, that it will also only be possible with the `bokeh_jupyter` extension installed (and possibly only in jupyterlab, not classic notbook)

**Comment by bryevdv** [comment]
> If I find this hook, what should it do? Generate a unique id? 

AFAIK currently Bokeh generates a div and a script with matching ids. The div is added to the output cell, and then the script is run (which embeds the plot in the div with the specified id). So offhand speculation is that: the new div in the new output cell needs to be updated, and the script needs to be updated to match.

**Comment by axil** [comment]
So ok, it is used to populate the proper div. Now the next question is: is the id used after the div has been populated (eg in `push_notebook` or whatever) or it is ok to just delete it afterwards?

**Comment by axil** [comment]
I'm also sort of working on jupyter integration ;) Almost finished an article with the description of the new lib. Would you mind having a look before it goes online? I expect it to be ready enough in a couple of days.

**Comment by bryevdv** [comment]
> So ok, it is used to populate the proper div. Now the next question is: is the id used after the div has been populated (eg in push_notebook or whatever) or it is ok to just delete it afterwards?

I am afraid I don't know the answer to that offhand. Bokeh is a complicated cross-runtime project, and Jupyter is a complicated cross-runtime project. Things will require some investigation to sort out just what is feasible.

> I'm also sort of working on jupyter integration ;) Almost finished an article with the description of the new wrapper lib. Would you mind having a look before it goes online? I expect it to be ready enough in a couple of days.

Sure, I can read an article, but please know that I am *far* from an expert on jupyter integration. To be completely honest, I am not much of a jupyter user at all. There are probably other people who can comment much better than I.

**Comment by axil** [comment]
Point me to a person who can :) @mattpap?

**Comment by philippjfr** [comment]
To be clear jupyterlite simply bypasses Tornado it does not attempt to actually run a tornado server. You'll find the discussion and various references/links here very relevant: https://github.com/holoviz/pyviz_comms/issues/3

Copying a cell and using output_view run into exactly the same problem.

**Comment by axil** [comment]
I tested push_notebook - it does not rely on id.

Do you have a community chat or meetings or something as I feel I'm somewhat offtopic here?

**Comment by bryevdv** [comment]
Some of us spend time on the Bokeh dev slack, you can request an invite here: https://slack-invite.bokeh.org/

But to reach the most people, it might make sense to convert this from and issue to a [Github Discussion](https://github.com/bokeh/bokeh/discussions) (or start a new one)
